internals-js/src/operations.ts

```diff
@@ -1574,12 +1575,12 @@ export class SelectionSet {
         ).setSelectionSet(minimizedSelectionSet);
 
         // Create a new "hash code" bucket or add to the existing one.
-        if (!equivalentSelectionSetCandidates) {
-          seenSelections.set(mockHashCode, [[selection.selectionSet, fragmentDefinition]]);
-          namedFragments.add(fragmentDefinition);
-        } else {
+        if (equivalentSelectionSetCandidates) {
           equivalentSelectionSetCandidates.push([selection.selectionSet, fragmentDefinition]);
+        } else {
+            seenSelections.set(mockHashCode, [[selection.selectionSet, fragmentDefinition]]);
         }
+        namedFragments.add(fragmentDefinition);
```

Comment:
If we get here, it's because we didn't find an exact match. In these cases, we need to create a fragment definition in all cases. Double check me here, but this does seem to fix the issue.

Reply:
Yeah, agree. Small nit, but could we bump this line up to where `fragmentDefinition` is assigned? Probably makes more sense right next to it.

Appreciate the if/else restructure 👍